metecho/api/models.py

```diff
@@ -915,6 +915,27 @@ class Task(
     assigned_dev = models.CharField(max_length=50, null=True, blank=True)
     assigned_qa = models.CharField(max_length=50, null=True, blank=True)
 
+    currently_refreshing_dataset_schema = models.BooleanField(default=False)
+    dataset_schema = models.JSONField(
+        default=dict,
+        encoder=DjangoJSONEncoder,
+        blank=True,
+        help_text=_("Cache of the Dev org schema related to this Task"),
+    )
+    currently_refreshing_datasets = models.BooleanField(default=False)
+    datasets = models.JSONField(
+        default=dict,
+        encoder=DjangoJSONEncoder,
+        blank=True,
+        help_text=_("Cache of the dataset definitions from the current Task branch"),
+    )
+    datasets_parse_errors = models.JSONField(
+        blank=True,
+        default=list,
+        help_text=_("User-facing errors that occurred during dataset refresh"),
+    )
+    currently_capturing_dataset = models.BooleanField(default=False)
```

Comment:
@jerivas If we move this to the `ScratchOrg` model, I think it could go away entirely and we could use `currently_capturing_changes` for both. Maybe let's rename that `currently_creating_commit` or `currently_retrieving_changes` or `currently_committing` or something?